docs/dev/structure.md

```diff
@@ -67,19 +74,34 @@ The aim of the scripts in here is to ensure that applications get upgraded acros
   - e.g. `lounge` has no "update" button in the application itself
   - however `radarr` has a built-in self-updater, so no `upgrade` script is necessary
 - A new major version needs to be opted-in by the user
-  - e.g. the Sonarr v2 to Sonarr v3 migration has to be made by user's choice, and the change requires a different apt package too
+  - e.g. the Sonarr v2 to Sonarr v3 migration has to be made by user's choice, and additional functionality needs to be executed to complete a succesful migration between the version.
 
 Before any of the upgrade scripts are ran, the `box update` procedure is done.
 
 ### Update
 `scripts/update/`
 
-**All scripts in this directory will be executed in alphabetical order when `box update` is ran**. 
+These scripts intend to make sure that the user's installation is up to spec with what we expect in our code, so that we do not need to keep track of all the different changes and the history of the package in all the scripts above.
+
+**All scripts in this directory will be executed in alphabetical order when `box update` is ran**. It is therefore required that scripts here have their "main functionality" wrapped in `if` statements which verify if the script should run.
+
+Each update statement is required to have the following implemented:
+- Conditional checks whether execution is necessary (exceptions apply, but discussed at PR-time)
+  - In general, this would usually consist of
+         1. Top-level check for whether an application is installed
+         2. Per-segment check whether the conditions for the update are present
+- A comment describing the condition for the update to trigger
+  - Any further context should be provided 
+- Handled echos
+  - `echo_info` for the start of an update segment and `echo_success` at the end of it
+  - An `echo_log` in case the update is being skipped. 
```

Comment:
These two are new rules I'd like to enforce here. There's a PR in the main repo which attempts at applying this to existing code.